e2e/wdio/core/tests/switch.e2e-spec.ts

```diff
@@ -0,0 +1,96 @@
+import {
+    click,
+    elementDisplayed,
+    getAttributeByName,
+    getElementArrayLength,
+    refreshPage,
+    waitForPresent
+} from '../../driver/wdio';
+import { switchPo } from '../../core/pages/switch.po';
+
+describe('Split-button test suite', () => {
+
+    const switchPage = new switchPo();
+    const {
+        switchSizes, switchSizesExample, toggle,
+        toggleInput, switchBindingExample, switchFormsExample, semanticswitchExample,
+        playGroundSwitchExample, disabledToggle, acceptIcon, declineIcon, semanticSwitch,
+        switchBtn, checkboxes, disabledAndCompactToggle
+    } = switchPage;
+
+    beforeAll(() => {
+        switchPage.open();
+    }, 1);
+
+    afterEach(() => {
+        refreshPage();
+        waitForPresent(switchSizes);
+    }, 1);
+
+   xit('Should check turn on/ Turn off switch toggle', () => {
+        checkSwitchingWork(switchSizesExample)
+        checkSwitchingWork(switchFormsExample)
+        checkSwitchingWork(semanticswitchExample)
+        checkSwitchingWork(switchBindingExample)
+        checkSwitchingWork(playGroundSwitchExample)
+    });
+
+    xit('Should check icons on semantic toggles', () => {
+        for (let i = 0; i < 2; i++) {
+            expect(elementDisplayed(declineIcon, i)).toEqual(true)
+            click(semanticSwitch, i)
+            expect(elementDisplayed(declineIcon, i)).toEqual(false)
+            expect(elementDisplayed(acceptIcon, i)).toEqual(true)
+        }
+    });
+
+    xit('Should check toggle state changes by click on buttons', () => {
+        click(switchBtn, 1)
+        expect(checkToggleState(switchBindingExample)).toEqual(true)
+        click(switchBtn, 1)
+        expect(checkToggleState(switchBindingExample)).toEqual(false)
+        click(switchBtn, 0)
+        expect(checkToggleState(switchBindingExample)).toEqual(true)
+        expect(checkToggleState(switchBindingExample)).toEqual(true)
+    });
+
+    it('should check switch toggle manage by checkboxes', () => {
+        click(checkboxes, 1)
+        expect(checkToggleState(playGroundSwitchExample)).toEqual(true)
+        
+        click(checkboxes, 0) 
+        expect(getAttributeByName(playGroundSwitchExample + toggle, 'class')).toEqual(disabledAndCompactToggle)
```

Comment:
should be able to use function: getElementClass(selector: string, index: number = 0) from wdio file.